sklearn/utils/estimator_checks.py

```diff
@@ -131,6 +131,29 @@ def _yield_classifier_checks(name, Classifier):
     if 'class_weight' in Classifier().get_params().keys():
         yield check_class_weight_classifiers
 
+def check_supervised_y_no_nan(name, Estimator):
+    # Checks that the Estimator targets are not NaN.
+
+    warnings.simplefilter("ignore")
+    X = np.random.randn(10, 5)
+    y1 = np.random.randn(10) / 0.
+    y2 = np.random.randn(10, 2) / 0.
+
+    errmsg = "Input contains NaN, infinity or a value too large for " \
+             "dtype('float64')."
+    try:
+        if "MultiTask" in name:
+            Estimator().fit(X, y2)
```

Comment:
Ironically if I didn't add that `if "MultiTask in name` statement, I wouldn't have discovered the bug fixed in a5860a1df21228989bbd8b5d601c18fc12e6550a. Would you still want me to use `multioutput_estimator_convert_y_2d`?